pkg/models/dashboard_queries.go

```diff
@@ -60,6 +60,17 @@ func GroupQueriesByPanelId(dashboard *simplejson.Json) map[int64][]*simplejson.J
 	return result
 }
 
+func HasExpressionQuery(queries []*simplejson.Json) bool {
+	for _, query := range queries {
+		uid := GetDataSourceUidFromJson(query)
+		if uid == "__expr__" {
```

Comment:
there is a `expr.IsDatasource(uid string) bool`

Reply:
Moved everything in this file to a model in the query service